src/main/java/com/squareup/kotlinpoet/ParameterSpec.kt

```diff
@@ -138,3 +138,23 @@ class ParameterSpec private constructor(builder: ParameterSpec.Builder) {
         = builder(name, type.asTypeName(), *modifiers)
   }
 }
+
+internal fun List<ParameterSpec>.emit(
+    codeWriter: CodeWriter,
+    emitBlock: (ParameterSpec) -> Unit = { it.emit(codeWriter) }
+) {
+
+  fun emit(prefix: String, separator: String, suffix: String) {
```

Comment:
Whoa, cool

Reply:
Though this might be simpler as just a when clause with cases for 0, 1, 2 and else!

Reply:
(no need to define prefix and suffix, each case could just be inline.)

Reply:
Makes sense! I pushed a08146c, gonna squash if nothing to change